pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -270,6 +271,33 @@ func (c *Cluster) startProvisioning(config *provisionConfig) {
 			}
 		}
 	}
+
+	// Deploying test based on the storageclassDeviceSet
+	// Todo: Replace the below logic to deploy osd pods over pvcs
+	for _, storageclassDeviceSet := range c.DesiredStorage.Storage {
+		for i := 0; i < storageclassDeviceSet.Count; i++ {
```

Comment:
You don't want to just loop over a count, you want to first check how many PVCs exist, and then create additional ones if there are not enough.

Reply:
yup added the logic to do it